### PR TITLE
[C-774] Move audius-backend to common, add native libs-instance

### PR DESCRIPTION
**Pull request by dylanjeffers**
### Description

- Moves AudiusBackend to audius-client common
- Add mobile audius-libs-instance,
- Add monitoringCallbacks to config since they rely on custom optimizely and analytics
- Add react-native-url-polyfill for libs monitoring callbacks (uses `new URL(...)`)

### How Has This Been Tested?

Able to make api requests through sdk in native

### CODE REVIEW

File: packages/mobile/src/services/audius-backend-instance.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    ethTokenBridgeAddress: Config.ETH_TOKEN_BRIDGE_ADDRESS,
+    solBridgeAddress: Config.SOL_BRIDGE_ADDRESS,
+    solTokenBridgeAddress: Config.SOL_TOKEN_BRIDGE_ADDRESS,
+    wormholeRpcHosts: Config.WORMHOLE_RPC_HOSTS
+  },
+  getLibs: async () => nativeLibs,
+  waitForLibsInit: async () => {},
+  withEagerOption: ({ normal }, ...args) => {
+    if (audiusLibs) {
+      console.log('eager option working?', audiusLibs)
```
Comment (sliptype) [inline packages/mobile/src/services/audius-backend-instance.ts:79]: Log, but probably fine for now

File: packages/mobile/src/services/audius-backend-instance.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+ * audiusBackend initialized for a mobile environment
+ */
+export const audiusBackendInstance = audiusBackend({
+  claimDistributionContractAddress: Config.CLAIM_DISTRIBUTION_CONTRACT_ADDRESS,
+  ethOwnerWallet: Config.ETH_OWNER_WALLET,
+  ethProviderUrls: (Config.ETH_PROVIDER_URL || '').split(','),
+  ethRegistryAddress: Config.ETH_REGISTRY_ADDRESS,
+  ethTokenAddress: Config.ETH_TOKEN_ADDRESS,
+  fetchCID: async () => ({}),
+  getFeatureEnabled: () => false,
```
Comment (sliptype) [inline packages/mobile/src/services/audius-backend-instance.ts:24]: We should be able to import remoteConfigInstance and get the feature flags that way, but stub is probably fine for now too

**Review by sliptype — APPROVED** [review]
Freaking amazing dude

File: packages/mobile/src/services/audius-backend-instance.ts (hunk 2 above)
Comment (dylanjeffers) [reply packages/mobile/src/services/audius-backend-instance.ts:24]: yeah agreed, so weirdly feature flags are broken on native when (when i modify metro to resolve native node_modules first, which we want ideally)

Comment (sliptype) [reply packages/mobile/src/services/audius-backend-instance.ts:24]: Interesting, does that have anything to do with the hooks or more likely the optimizely sdk?

Comment (dylanjeffers) [reply packages/mobile/src/services/audius-backend-instance.ts:24]: still investigating, testing to see if its related to polyfills, wether we load the web vs native optimizely code etc...

Comment (dylanjeffers) [reply packages/mobile/src/services/audius-backend-instance.ts]: yay figured it out, added.

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/dj-c-774-common-audius-backend

**Pull request merged**